### ISSUE TITLE
Registering multiple filters

### ISSUE THREAD

**Issue by chris-barnett** [nervgh/angular-file-upload#297]
How?

Have tried:

$scope.uploader.filters.push({

```
            name: 'PhotoFilter',
            fn: function (item /*{File|FileLikeObject}*/, options) {
                //$scope.errorMsg = "";
                var type = '|' + item.type.slice(item.type.lastIndexOf('/') + 1) + '|';
                var x = '|jpg|png|jpeg|bmp|gif|'.indexOf(type) !== -1;
                if (!x) $scope.errorMsg = "Incorrect file format! This will not be uploaded. Please try again.";
                return x;
            },

            name: 'VideoFilter',
            fn: function (item /*{File|FileLikeObject}*/, options) {
                $scope.errorMsg = "";
                var type = '|' + item.type.slice(item.type.lastIndexOf('/') + 1) + '|';
                var x = '|mp4|mpeg|wmv|mov|avi|'.indexOf(type) !== -1;
                if (!x) $scope.errorMsg = "Incorrect file format! This will not be uploaded. Please try again.";
                return x;
            }
        });
```

This only seems to register the last Filter?

**Comment by sunild** [comment]
@fabiokiatkowski It's [perfectly acceptable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push) to use Array.push() and pass in multiple items. This has been working in my app for a long time.

What is strange, however, is that I recently changed the name of a filter, and it stopped being called. I can't seem to figure out what the issue is from the source.

**Comment by sunild** [comment]
Registering multiple filters as shown above with `uploader.filters.push(filter1, filter2)` [works just fine](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/push). I'm using code as shown above in my app.

However, I ended up here because when I changed the name of one of my filters recently, it stopped working.

A coworker just pointed out that was because in the HTML we reference the filter names:

`<input id="file-selector" type="file" nv-file-select uploader="builder.uploader" filters="imageFilter,canAddFileFilter">`

And so I needed to change the markup to match the new filter name.

@nervgh I think this issue can be closed, as the code in the original comment works. Perhaps the OP had a similar issue as mine.
